homeassistant/components/ecobee/switch.py

```diff
@@ -0,0 +1,85 @@
+"""Switch entity for ecobee."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.climate import HVACMode
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import _LOGGER, DOMAIN, ECOBEE_AUX_HEAT_ONLY
+
+AUX_HEAT_ONLY_KEY = "aux_heat_only"
+
+SWITCH_TYPES: tuple[SwitchEntityDescription, ...] = (
+    SwitchEntityDescription(
+        key=AUX_HEAT_ONLY_KEY,
+        name=AUX_HEAT_ONLY_KEY,
+        translation_key=AUX_HEAT_ONLY_KEY,
+        device_class=SwitchDeviceClass.SWITCH,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Ecobee switches."""
+    data = hass.data[DOMAIN]
+
+    async_add_entities(
+        EcobeeSwitch(data, index, description)
+        for index in range(len(data.ecobee.thermostats))
+        # if data.ecobee.get_thermostat(index)["settings"]["hasHeatPump"]
+        for description in SWITCH_TYPES
+    )
+
+
+class EcobeeSwitch(SwitchEntity):
```

Comment:
I think perhaps it would be good to make a base entity in a preliminary PR and then add it here to reduce duplication

Reply:
I've created a base switch entity and a auxHeat switch inheriting from the base.

Reply:
Refacotred based on newly added switch code.